src/Sylius/Behat/Context/Ui/UserContext.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Behat\Context\Ui;
+
+use Behat\Behat\Context\Context;
+use Sylius\Behat\Page\Customer\CustomerShowPage;
+use Sylius\Behat\Page\Customer\CustomersIndexPage;
+use Sylius\Component\Core\Test\Services\SharedStorageInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+/**
+ * @author Magdalena Banasiak <magdalena.banasiak@lakion.com>
+ */
+class UserContext implements Context
+{
+    /**
+     * @var SharedStorageInterface
+     */
+    private $sharedStorage;
+
+    /**
+     * @var RepositoryInterface
+     */
+    private $customerRepository;
+
+    /**
+     * @var CustomersIndexPage
+     */
+    private $customersIndexPage;
+
+    /**
+     * @var CustomerShowPage
+     */
+    private $customerShowPage;
+
+    /**
+     * @param SharedStorageInterface $sharedStorage
+     * @param RepositoryInterface $customerRepository
+     * @param CustomersIndexPage $customersIndexPage
+     * @param CustomerShowPage $customerShowPage
+     */
+    public function __construct(
+        SharedStorageInterface $sharedStorage,
+        RepositoryInterface $customerRepository,
+        CustomersIndexPage $customersIndexPage,
+        CustomerShowPage $customerShowPage
+    ) {
+        $this->sharedStorage = $sharedStorage;
+        $this->customerRepository = $customerRepository;
+        $this->customersIndexPage = $customersIndexPage;
+        $this->customerShowPage = $customerShowPage;
+    }
+
+    /**
+     * @When I delete the account of :email
+     */
+    public function iDeleteAccount($email)
+    {
+        $customersIndexPage = $this->customersIndexPage;
+        $customersIndexPage->open();
+        $customersIndexPage->deleteUser($email);
```

Comment:
The delete should be from `customerDetailsPage` since we support validation and if there will be more than 10 users you may not find this user on the first page.